core/include/debug.h

```diff
@@ -52,12 +52,7 @@ extern "C" {
 #include "cpu_conf.h"
 #define DEBUG_PRINT(...) \
     do { \
-        if ((sched_active_thread == NULL) || (sched_active_thread->stack_size > THREAD_EXTRA_STACKSIZE_PRINTF)) { \
```

Comment:
looks unrelated?

Reply:
@haukepetersen Can you verify this?

Reply:
This is indeed unrelated. I actually don't remember the reason for this if here (except that is mostly very annoying), but I think there were platforms that could not handle printf in interrupt context -> which poses the question again, why the custom printf is not executed, but instead a puts with a fairly long string is... But this discussion is independent of this PR, so just take this change out.